src/core.jl

```diff
@@ -151,10 +151,16 @@ function TensorShape(dims::Vector)
     TensorShape([x<0 ? Nullable{Int64}() : Nullable{Int64}(x) for x in dims])
 end
 
-function TensorShape(dim::Void)
+function TensorShape(::typeof(collect(tuple())))
```

Comment:
Yes, could instead say: `Vector{Union{}}`.

I was playing with the idea, that `(::typeof(collect(tuple())` might be clearer.
Since it does not require remembering that collecting some untyped (not Any-typed or Void typed) collection gives `Vector{Union{}}`.

I will make the change